src/app/pages/confirm-test-details/confirm-test-details.page.ts

```diff
@@ -1,34 +1,253 @@
+          select(getTestOutcomeText),
+        ),
+        activityCode$: currentTest$.pipe(
+          select(getActivityCode),
+        ),
+        testCategory$: currentTest$.pipe(
+          select(getTestCategory),
+          map((testCategory) => testCategory as TestCategory),
+        ),
+        transmission$: currentTest$.pipe(
+          select(vehicleDetails.vehicleDetails),
+          select(getGearboxCategory),
+        ),
+        code78$: currentTest$.pipe(
+          select(getPassCompletion),
+          select(getCode78),
+        ),
+        d255$: currentTest$.pipe(
+          select(getTestSummary),
+          select(getD255),
+        ),
+      };
+      if (category !== TestCategory.ADI2) {
+        this.pageState = {
+          ...this.pageState,
+          provisionalLicense$: currentTest$.pipe(
+            select(getPassCompletion),
+            map(isProvisionalLicenseProvided),
+          ),
+        };
+      }
+    });
+
+    const {
+      testCategory$, testOutcomeText$, candidateUntitledName$, slotId$,
+    } = this.pageState;
+
+    this.merged$ = merge(
+      testCategory$.pipe(map((value) => this.category = value)),
+      testOutcomeText$.pipe(map((value) => this.testOutcome = value)),
+      candidateUntitledName$.pipe(tap((value) => this.candidateName = value)),
+      slotId$.pipe(map((slotId) => this.slotId = slotId)),
+    );
+  }
+
+  isTerminated(testResult: string): boolean {
+    return testResult === TestOutcome.Terminated;
+  }
+
+  isPassed(testResult: string): boolean {
+    return testResult === TestOutcome.Passed;
+  }
+
+  getActivityCode(activityCodeModel: ActivityCodeModel): ActivityCode {
+    return activityCodeModel.activityCode;
+  }
+
+  getProvisionalText(received: boolean): LicenceReceivedText {
+    return received ? LicenceReceivedText.TRUE : LicenceReceivedText.FALSE;
+  }
+
+  getD255Text(d255: boolean): D255 {
+    return d255 ? D255.TRUE : D255.FALSE;
+  }
 
-  ngOnInit() {
+  async onSubmit() {
+    await this.showConfirmTestDetailsModal();
   }
 
-  navigateBack(): void {
-    this.navController.back();
+  async showConfirmTestDetailsModal(): Promise<void> {
+    const alert = await this.alertController.create({
+      message: `You are about to submit a ${this.testOutcome} Cat ${this.category} test for ${this.candidateName}
+                <br/><br/>Are you sure you want to submit this result?`,
+      cssClass: 'confirm-declaration-modal',
+      buttons: [
+        {
+          text: 'Cancel',
+          handler: () => {
+          },
+        },
+        {
+          text: 'Submit',
+          handler: () => this.onTestDetailsConfirm(),
+        },
+      ],
+      backdropDismiss: false,
+    });
+    await alert.present();
   }
 
-  async navigateDebrief(): Promise<void> {
-    await this.router.navigate([TestFlowPageNames.DEBRIEF_PAGE]);
+  async onTestDetailsConfirm(): Promise<void> {
+    this.store$.dispatch(SetTestStatusWriteUp(this.slotId));
```

Comment:
The dispatching of actions are wrapped in a condition on DES3, should this be the same here?